### DISCUSSION TITLE
UNet PonyXL Deep Dive - Testing WF Attached

### DISCUSSION THREAD

**Discussion by Kinglord** [cubiq/prompt_injection#8] [General]
Hello,

First off super excited about this node and approach, since I've been working with Pony for a project I decided to take on the task of seeing what I could find out in regards to how bad things are under the covers. I can say that it's about as gnarly as I thought, but hopefully this can still be of use even with something as trained as PDXL is. I've attached the ComfyUI WF I've been using for testing along with some notes. I'll post my UNet findings here as well and would love help debunking some I got wrong, or figuring out how some of these play together. I think the 2 big things to understand at this point is what blocks need partner blocks to perform better with Pony, and what strength works best when tokens need to be split - i.e. sending the same token to one block a STR 1, another at .5, etc.

Feel free to hit me up here or on Discord @ kinglord (I'm active there way more than GH) and feel free to share this out to anyone else up for experimenting with and getting into the weeds on Pony Blocks.

My hope is we can figure out the best thing to do with various tokens that hit across multiple block levels to better control results - best example I've hit so far is something like "muscular." It hits Input 8, Output 0 and Output 1 all strongly but in different ways. I8 does drastic character changes, O0 is focused more on the face, and O1 drives muscle tone for the body, so I think ideally it's isolated to Output 1 if you're trying to just tone someone up.

PONYXL TESTING WF
[ponyAttnTesting.json](https://github.com/user-attachments/files/15794127/ponyAttnTesting.json)

### PonyXL UNet Block Notes - WIP AND MIGHT BE WRONG ###
**_NOTES - Pony seems to super overload Input 8 and Output 1, I'm hoping we kind find ways to work around or improve this. Inputs 4/5 and Outputs 4/5 all feel weak in influence and I feel like there's something I'm missing here in regards to details/colors?_**
- Input 4
  - Weak
    Makeup / Skin Tone Tokens
    Overall "beauty" aesthetic, maybe ratios?
- Input 5
  - Weak
  Highlights / Lighting, Line Details
  Character Tokens (Ethnicity, Celebs, Etc.)
- Input 7
Body Type / Proportion Tokens
Eye Tokens
Outfit / Clothes Tokens
Camera Tokens (close-up, full body)
  - Weak
  Hair Style Tokens
  Style Tokens
- Input 8
Gender Tokens
Body Type / Proportion Tokens
Style Tokens
Hair Style Tokens
Hair Color Tokens
Eye Tokens
Outfit / Clothes Tokens
Pose Tokens
Background Tokens
Camera Tokens (close-up, full body)
- Middle 0
Expression Tokens
Eye Tokens
  - Weak
  Hair Style Tokens
  Outfit / Clothes Tokens
  Body Type / Proportion Tokens
  Also seems to smooth?
- Output 0
Gender Tokens
Face Tokens
Hair Style Tokens
Hand Tokens
Lighting
Face / Body details (like ears, breasts)?
  - Weak
  Character Tokens (Ethnicity, Celebs, Etc.)
  Hair Style Tokens
  Body Type / Proportion Tokens
  Style Tokens
- Output 1
Pony Scores & Source Tokens
Style Tokens
Character Tokens (Ethnicity, Celebs, Etc.)
Face Tokens
Eye Tokens
Expression Tokens
Hand Tokens
Makeup / Skin Tone Tokens
Hair Color Tokens
Background Tokens
  - Weak
  Outfit / Clothes Tokens
- Output 2
Mouth Tokens
  - Weak
  Expression Tokens
  Textures?
- Output 3
Expression Tokens
Eyes Tokens
Face Detail Tokens
  - Weak
  Mouth Tokens
  Expression Tokens
  Hair Details
  Clothing Details
  Accessories
- Output 4
  - Weak
  Lighting / Wind? & Motion?
Pose Tokens
- Output 5
  - Weak
  Lighting, Fine details, Edges, Accents

**Comment by cubiq** [comment]
wow this is a deep dive into the rabbit hole! Thanks for sharing.

remember that you can lower the weight of a block with a simple (tokens:0.5) syntax if in8 and out1 are too strong.
